bundle/edu.gemini.pot/src/test/scala/edu/gemini/pot/sp/ProgramGen.scala

```diff
@@ -136,6 +166,17 @@ object ProgramGen {
       n.title = decorativeTitle(n, title)
     }
 
+  val genEditSplitTemplateGroup: Gen[ProgEdit] =
+    for {
+      tgf <- maybePickTemplateGroup
+    } yield { (f: ISPFactory, p: ISPProgram) =>
```

Comment:
`tgf` stands for "template group function" here.  It's the function that when applied with a program will select a random `ISPTemplateGroup` from the program, if there is at least one.  The `for` expression in turn just yields a function that when provided with a program factory and a program, will split a random template group in the program, if there is one.

Unfortunately everything is awkward like this because of the mutable program model.  The generator doesn't actually mutate anything, it just generates functions that when handed a program do the actual mutation.